Dockerfile

```
@@ -0,0 +1,54 @@
+# GPTQ-for-LLaMa and Text Generation WebUI Dockerfile
+FROM nvidia/cuda:11.8.0-devel-ubuntu22.04 as builder
+
+RUN apt-get update && \
+    apt-get install --no-install-recommends -y git build-essential python3-dev python3-pip && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN --mount=type=cache,target=/root/.cache/pip pip3 install torch torchvision torchaudio
+RUN git clone https://github.com/qwopqwop200/GPTQ-for-LLaMa /build
+
+WORKDIR /build
+
+ARG GPTQ_VERSION
+RUN git checkout ${GPTQ_VERSION}
+
+RUN --mount=type=cache,target=/root/.cache/pip pip3 install -r requirements.txt
+
+# https://developer.nvidia.com/cuda-gpus
+# for a rtx 2060: ARG TORCH_CUDA_ARCH_LIST="7.5"
+ARG TORCH_CUDA_ARCH_LIST="3.5;5.0;6.0;6.1;7.0;7.5;8.0;8.6+PTX"
+RUN python3 setup_cuda.py bdist_wheel -d .
+
+FROM nvidia/cuda:11.8.0-runtime-ubuntu22.04
+
+LABEL maintainer="Your Name <your.email@example.com>"
+LABEL description="Docker image for GPTQ-for-LLaMa and Text Generation WebUI"
+
+RUN apt-get update && \
+    apt-get install --no-install-recommends -y git python3 python3-pip && \
+    rm -rf /var/lib/apt/lists/*
+
+RUN --mount=type=cache,target=/root/.cache/pip pip3 install torch torchvision torchaudio
```

```suggestion
RUN --mount=type=cache,target=/root/.cache/pip pip3 install --upgrade pip setuptools
RUN --mount=type=cache,target=/root/.cache/pip pip3 install torch torchvision torchaudio
```

Same argument above.

---

but this is the second stage